app/models/user.rb

```diff
@@ -61,4 +61,96 @@ def generate_token(column)
       break unless User.exists?(column => self[column])
     end
   end
+
+  # @return [TrueClass, FalseClass]
+  def uploaded_before? = !user_resources.all.empty?
+
+  # noinspection RubyNilAnalysis
+  def last_uploaded_file
+    return 'none' unless uploaded_before?
+
+    user_resources.last.name
+  end
+
+  # @return Array[Hash]
+  def upload_file_history
+    upload_file_history = []
+    all_upload_dates.each do |item|
+      upload_file_history.append({ 'date': item, 'number': get_number_of_files_at_date(item) })
+    end
+    upload_file_history
+  end
+
+  # @return Array[Hash]
+  # rubocop:disable Metrics/MethodLength
+  def number_of_files_per_type
+    hash_array = []
+    types = {}
+    user_resources.each do |item|
+      if types.include? item.type.to_s
+        types[item.type.to_s] += 1
+      else
+        types[item.type.to_s] = 1
+      end
+    end
+
+    types.each do |item|
+      hash_array.append({ 'type': item[0], 'number': item[1] })
+    end
+    hash_array
+  end
+
+  # rubocop:enable Metrics/MethodLength
+
+  # @return Array[Hash]
+  def top_five_files_by_size
+    hash_array = []
+    user_resources.each do |item|
+      hash_array.append({ 'file': item.name, 'size': item.byte_filesize })
+    end
+    hash_array.sort! { |a, b| a[:size] <=> b[:size] }
+    if hash_array.size >= 5
+      hash_array.reverse.first 5
+    else
+      hash_array.reverse
+    end
+  end
+
+  def times_of_login
+    login_counter
+  end
+
+  def total_number_of_uploads
+    user_resources.count
+  end
+
+  def total_upload_size
+    total_upload_size = 0
+    user_resources.each do |item|
+      total_upload_size += item.byte_filesize
+    end
+    total_upload_size
+  end
+
+  private
+
+  # noinspection RubyNilAnalysis
+  # @return [Array]
+  def all_upload_dates
+    all_upload_dates = []
+    user_resources.each do |item|
+      all_upload_dates.append(item.creation_date)
+    end
+    all_upload_dates.uniq
```

Comment:
Auch hier gibt es schon effizientere und fertige Methoden in AR. So in der Art wie:

`all_upload_dates.group(:creation_date).pluck(:creation_date)`

https://apidock.com/rails/ActiveRecord/Calculations/pluck
https://edgeguides.rubyonrails.org/active_record_querying.html#selecting-specific-fields